btf/types.go

```diff
@@ -67,13 +71,73 @@ func (ts types) ByID(id TypeID) (Type, error) {
 	return ts[id], nil
 }
 
+type Skipper func(Type) bool
+
+type Comparer func(Type, Type) bool
+
+// SkipQualifierAndTypedef skips qualifiers and Typedefs.
+var SkipQualifierAndTypedef = func(t Type) bool {
+	switch t.(type) {
+	case qualifier, *Typedef:
+		return true
+	}
+	return false
+}
+
+// CompareTypeName returns true if a and b's names correspond.
+var CompareTypeName = func(a, b Type) bool {
+	return a.TypeName() == b.TypeName()
+}
+
+// Equal compares Type a to Type b. By default, only binary equivalency is
+// checked, ignoring type names.
+//
+// cmp allows providing extra comparison rules. It will be evaluated after
```

Comment:
I think the semantics should be: if cmp == nil we default to an exported function that does what you call binary equivalency.  Otherwise we defer to cmp entirely.

That way we could use this function to replace `coreAreTypesCompatible` which has a completely different notion of equality.